### PR TITLE
feat (define-tag-after-class-definition): new rule

### PR DESCRIPTION
**Pull request by 43081j**
Can do the docs later for each of these rules

This one catches classes which haven't been defined at all or were defined before their associated class

### CODE REVIEW

File: src/rules/define-tag-after-class-definition.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+            if (ref?.resolved && ref.resolved.defs.length === 1) {
+              seenClasses.delete(ref.resolved.defs[0].node);
+            }
+          } else if (tagClass.type === 'ClassExpression') {
+            seenClasses.delete(tagClass);
+            context.report({
+              node: tagClass,
+              messageId: 'noExpressions'
+            });
```
Comment (keithamus) [inline src/rules/define-tag-after-class-definition.ts:72]: It looks like this branch lacks test coverage.

**Review by keithamus — APPROVED** [review]
LGTM

File: src/rules/define-tag-after-class-definition.ts (hunk 1 above)
Comment (43081j) [reply src/rules/define-tag-after-class-definition.ts:72]: i think this one tests it:

```ts
    {
      code: `customElements.define('x-foo', class extends HTMLElement {
      });`,
      errors: [
        {
          messageId: 'noExpressions',
          line: 1,
          column: 32
        }
      ]
    },
```

does that make sense?

Comment (keithamus) [reply src/rules/define-tag-after-class-definition.ts:72]: Yep! Thanks

**Pull request merged**